yt_dlp/extractor/tiktok.py

```diff
@@ -911,3 +911,35 @@ def _real_extract(self, url):
         if self.suitable(new_url):  # Prevent infinite loop in case redirect fails
             raise UnsupportedError(new_url)
         return self.url_result(new_url)
+
+
+class TikTokLiveIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?tiktok\.com/@(?P<id>[\w\.-]+)/live'
+    IE_NAME = 'tiktok:live'
+
+    def _real_extract(self, url):
+        uploader = self._match_id(url)
+        webpage = self._download_webpage(url, uploader, headers={'User-Agent': 'User-Agent:Mozilla/5.0'})
+        room_id = self._html_search_regex(r'snssdk\d*://live\?room_id=(\d+)', webpage, 'room ID', default=None)
+        if not room_id:
+            raise ExtractorError('The user is not currently live', expected=True)
+        video_js_data = self._download_json(
+            f'https://www.tiktok.com/api/live/detail/?aid=1988&roomID={room_id}', room_id)
```

Comment:
this is a bit cleaner and will be easier to maintain if we need to add more query params in the future
```suggestion
        video_js_data = self._download_json(
            'https://www.tiktok.com/api/live/detail/', room_id, query={
                'aid': '1988',
                'roomID': room_id,
            })
```